engine/runtime/src/main/java/org/enso/compiler/TreeToIr.java

```diff
@@ -1879,26 +1887,28 @@ def translateInvalid(invalid: AST.Invalid): Expression = {
     * Currently this only supports documentation comments, and not standarc
     * types of comments as they can't currently be represented.
     *
-    * @param tree the comment to transform
+    * @param doc the comment to transform
     * @return the [[IR]] representation of `comment`
     */
-  IR.Comment translateComment(Tree tree) {
-    return switch (tree) {
-      case Tree.ArgumentBlockApplication comment -> {
-        var doc = new StringBuilder();
-        var sep = "";
-        for (var l : comment.getArguments()) {
-          if (l.getExpression() != null) {
-            doc.append(sep);
-            doc.append(l.getExpression().codeRepr());
-            sep = "\n";
+  IR.Comment translateComment(Tree.Documented doc) {
+      var msg = new StringBuilder();
+      for (var t : doc.getElements()) {
+        switch (t) {
+          case TextElement.Section s -> {
+            var txt = s.getText().codeRepr();
+            if (txt.startsWith("#")) {
```

Comment:
This is pretty ugly. However the new parser gives me `#` while the old one doesn't do that for:
```
## Selects columns based on their names.
```
CCing @kazcw .